src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -304,6 +307,25 @@ public void transferItemHold(UUID transactionId, String itemBarcode) {
     eventPublisher.publishEvent(MoveRequestEvent.of(transaction, item));
   }
 
+  @Override
+  public InnReachTransactionDTO transferLocalHold(UUID transactionId, String itemBarcode) {
+    var transaction = fetchTransactionOfType(transactionId, LOCAL);
+    var hold = transaction.getHold();
+    var item = fetchItemByBarcode(itemBarcode);
+
+    transaction.setState(TRANSFER);
+    hold.setFolioItemId(item.getId());
+
+    var holdingId = item.getHoldingsRecordId();
+    if (!hold.getFolioHoldingId().equals(holdingId)) {
+      transaction.getHold().setFolioHoldingId(holdingId);
+      var holding = fetchHoldingById(holdingId);
+      transaction.getHold().setFolioInstanceId(holding.getInstanceId());
+    }
+
+    return transactionMapper.toDTO(transaction);
```

Comment:
I guess we should also check if the request (associated with transaction by folioRequestId) is linked to the item. If it's not then we should trigger RequestService.moveItemRequest

Reply:
Also, there is a change in requirements for similar story handling Item Holds - it should accept item ID instead of barcode as a path variable. I assume it applies here as well